dipy/reconst/dki.py

```diff
@@ -0,0 +1,1606 @@
+    # Reshape data according to the shape of dki_params
+    pred_sig = pred_sig.reshape(dki_params.shape + len(pred_sig))
+
+    return pred_sig
+
+
+class DKIModel(ReconstModel):
+    """ Diffusion Kurtosis Tensor
+    """
+    def __init__(self, gtab, fit_method="OLS_DKI", *args, **kwargs):
+        """ Diffusion Kurtosis Tensor Model [1]
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+
+        fit_method : str or callable
+            str can be one of the following:
+            'OLS_DKI' or 'ULLS_DKI' for ordinary least squares
+                dki.ols_fit_dki
+            'WLS_DKI' or 'UWLLS_DKI' for weighted ordinary least squares
+                dki.wls_fit_dki
+
+            callable has to have the signature:
+                fit_method(design_matrix, data, *args, **kwargs)
+
+        args, kwargs : arguments and key-word arguments passed to the
+           fit_method. See dki.ols_fit_dki, dki.wls_fit_dki for details
+
+        References
+        ----------
+           [1] Tabesh, A., Jensen, J.H., Ardekani, B.A., Helpern, J.A., 2011.
+           Estimation of tensors and tensor-derived measures in diffusional
+           kurtosis imaging. Magn Reson Med. 65(3), 823-836
+        """
+        ReconstModel.__init__(self, gtab)
+
+        if not callable(fit_method):
+            try:
+                self.fit_method = common_fit_methods[fit_method]
+            except KeyError:
+                raise ValueError('"' + str(fit_method) + '" is not a known fit '
+                                 'method, the fit method should either be a '
+                                 'function or one of the common fit methods')
+
+        self.design_matrix = dki_design_matrix(self.gtab)
+        self.args = args
+        self.kwargs = kwargs
+
+
+    def fit(self, data, mask=None):
+        """ Fit method of the DKI model class
+
+        Parameters
+        ----------
+        data : array
+            The measured signal from one voxel.
+
+        mask : array
+            A boolean array used to mark the coordinates in the data that
+            should be analyzed that has the shape data.shape[-1]
+        """
+        # If a mask is provided, we will use it to access the data
+        if mask is not None:
+            # Make sure it's boolean, so that it can be used to mask
+            mask = np.array(mask, dtype=bool, copy=False)
+            data_in_mask = data[mask]
+        else:
+            data_in_mask = data
+
+        params_in_mask = self.fit_method(self.design_matrix, data_in_mask,
+                                         *self.args, **self.kwargs)
+
+        if mask is None:
+            out_shape = data.shape[:-1] + (-1, )
+            dki_params = params_in_mask.reshape(out_shape)
+        else:
+            dki_params = np.zeros(data.shape[:-1] + (27,))
+            dki_params[mask, :] = params_in_mask
+
+        return DKIFit(self, dki_params)
+
+    def predict(self, dki_params, S0=1):
```

Comment:
The predict method should take an optional gtab argument so that predictions can be made on arbitrary gradient sets.